drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -4238,6 +4241,47 @@ static struct notifier_block netevent_notifier = {
 	.notifier_call = rswitch_netevent_cb,
 };
 
+static ssize_t l3_offload_show(struct device *dev, struct device_attribute *attr,
+			       char *buf)
+{
+	return sysfs_emit(buf, "%d\n", glob_priv->offload_enabled);
+}
+
+static void rswitch_disable_offload(struct rswitch_private *priv)
+{
+	struct rswitch_device *rdev;
+
+	read_lock(&priv->rdev_list_lock);
+	list_for_each_entry(rdev, &priv->rdev_list, list)
+		cleanup_all_routes(rdev);
+	read_unlock(&priv->rdev_list_lock);
+}
+
+static ssize_t l3_offload_store(struct device *dev, struct device_attribute *attr,
+				const char *buf, size_t count)
+{
+	long new_value;
+
+	if (kstrtol(buf, 10, &new_value))
+		return -EINVAL;
+
+	new_value = !!new_value;
+	if (new_value != glob_priv->offload_enabled) {
+		if (new_value) {
+			register_fib_notifier(&init_net, &glob_priv->fib_nb, NULL, NULL);
+		} else {
+			rswitch_disable_offload(glob_priv);
+			unregister_fib_notifier(&init_net, &glob_priv->fib_nb);
+		}
+
+		glob_priv->offload_enabled = new_value;
```

Comment:
I think this should come first and be covered with write_lock, to prevent learning chain from adding new routes after `rswitch_disable_offload`

Reply:
It won't happen because after rswitch_disable_offload is executed, the list with routes becomes empty, so learning chain won't add new entries.